pkg/apis/componentconfig/types.go

```diff
@@ -596,6 +596,15 @@ type KubeSchedulerConfiguration struct {
 	FailureDomains string
 	// leaderElection defines the configuration of leader election client.
 	LeaderElection LeaderElectionConfiguration
+	// PolicyConfigMapName is the name of the ConfigMap object that specifies
+	// the scheduler's policy config. If UseLegacyPolicyConfig is true, scheduler
+	// uses PolicyConfigFile. If UseLegacyPolicyConfig is false and
+	// PolicyConfigMapName is not empty, the ConfigMap object with this name must
+	// exist before scheduler initialization.
+	PolicyConfigMapName string
```

Comment:
If our component config reduces to a config map, should all other params shuffle to config map and be removed from the component config?  

Also we need a means to generate a config map with all the defaults.